chain-vote/src/tally.rs

```diff
@@ -381,55 +389,47 @@ impl Tally {
     }
 }
 
-pub fn batch_decrypt<'a, T, I>(
-    encrypted_tallies: T,
-    member_pks: &[MemberPublicKey],
-    member_sks: &[&MemberSecretKey],
-) -> Result<Vec<Tally>, TallyError>
-where
-    T: IntoIterator<Item = &'a EncryptedTally>,
-    T::IntoIter: Clone,
-{
-    let encrypted_tallies = encrypted_tallies.into_iter();
-    let absolute_max_stake = encrypted_tallies
-        .clone()
-        .map(|tally| tally.max_stake)
-        .max()
-        .map(TryInto::try_into);
-
-    match absolute_max_stake {
-        Some(Ok(absolute_max_stake)) => {
+pub fn batch_decrypt(validated_tallies: &[ValidatedTally]) -> Result<Vec<Tally>, TallyError> {
+    let absolute_max_stake = validated_tallies.iter().map(|tally| tally.max_stake).max();
+
+    let absolute_max_stake = match absolute_max_stake {
+        Some(stake) => stake,
+        None => return Ok(vec![]),
+    };
+
+    match absolute_max_stake.try_into() {
```

Comment:
nit: I would say it's a bit less readable to have this `try_into` without explicit type (which I guess is `NonZeroU64`) compared to match for 0 and non zero like
```
match absolute_max_stake {
    0 => 
    non_zero => Table::generate(non_zero.try_into().unwrap())
```

Reply:
I guess I went for this approach to avoid the unwrap call. What's your preference between the following:
```rust
match absolute_max_stake {
   0 => {},
  non_zero => Table::generate(non_zero.try_into().unwrap()),
}
// or
match NonZeroU64::try_from(absolute_max_stake) {
  Err(_) => },
  Ok(non_zero) => Table::generate(non_zero),
}
```

Reply:
Personally probably the first, but it doesn't really matter. Having the explicit cast is already a good step forward, whatever you prefer between these two is fine